scripts/core/editor3/components/Editor3.jsx

```diff
@@ -94,12 +94,17 @@ export class Editor3Component extends React.Component {
     /**
      * @ngdoc method
      * @name Editor3#onDragOver
-     * @returns {Boolean} Returns true if the item is permitted.
+     * @returns {Boolean} Returns true if the item is not permitted.
      * @description Checks if the dragged over item is allowed.
      */
     onDragOver(e) {
+        const {editorFormat, readOnly, singleLine} = this.props;
         const mediaType = e.originalEvent.dataTransfer.types[0] || '';
 
+        if (readOnly || singleLine || !(editorFormat && editorFormat.indexOf('picture') !== -1)) {
```

Comment:
Editor3 component don't have any default value yet, but yes it is a good idea to add them now. 
Sometimes very long conditions are difficult to read and understood, for example your proposed solution is wrong, it should be || instead &&  (return !readOnly && editorFormat.indexOf('picture') > -1 || ....). Probably a third solution will be better.
Related to singleLine I think it is more stable the fact that a single line don't contain an image than the fact that a singleLine don't support any formatting. Also the singleLine condition is more clear.
I think the current tests covers all cases for onDragOver.